Sources/PrayerTimes.swift

```diff
@@ -137,6 +137,16 @@ public struct PrayerTimes {
                 tempIsha = safeIsha
             }
         }
+        
+        // Maghrib calculation with check against safe value
+        if let maghribAngle = calculationParameters.maghribAngle,
+            let maghribComponents = solarTime.timeForSolarAngle(Angle(-maghribAngle), afterTransit: true),
+            let maghribDate = cal.date(from: maghribComponents),
+            // maghrib safe if falls between asr and maghrib
+            (tempAsr?.compare(maghribDate) == .orderedAscending || tempAsr == nil),
+            (tempIsha?.compare(maghribDate) == .orderedDescending || tempIsha == nil) {
+                tempMaghrib = maghribDate
+        }
```

Comment:
Good call on getting rid of `tempSunset` and refactoring this.

It's hard to test the edge cases, so here's the original code to compare next to:

```
        // Maghrib calculation with check against safe value
        if let maghribAngle = calculationParameters.maghribAngle {
            if let maghribComponents = solarTime.timeForSolarAngle(Angle(-maghribAngle), afterTransit: true) {
                tempMaghrib = cal.date(from: maghribComponents)
            }
            
            if let tempAsr = tempAsr, let tempIsha = tempIsha {
                // maghrib safe if falls between asr and maghrib
                if tempMaghrib == nil || tempMaghrib?.compare(tempAsr) == .orderedAscending || tempMaghrib?.compare(tempIsha) == .orderedDescending {
                    tempMaghrib = tempSunset
                }
            } else {
                // fallback to regular sunset
                tempMaghrib = tempSunset
            }
        } else {
            tempMaghrib = tempSunset
        }
```

Reply:
Does a negative Maghrib angle make sense, or do I have to still check that it's greater than zero?

Reply:
In reality we are forcing it to negative `Angle(-maghribAngle)`, because you're looking for an angle below the horizon, if they provided a negative value and then we forced it to positive it would be above the horizon and produce a time earlier than sunset. 

If we can better define what the true values are, we can have a better sanity check.

Reply:
If we can find better information about what this is supposed to do that would help, it would seem Jafari maghrib probably requires it to be post sunset, so maybe a better comparison is between sunset and Isha.

Also in Swift `Date` now conforms to `Comparable` so we don't need to use the `compare()` function, we can just use the `<` and `>` operators.

Reply:
I'll find out more answers about this. For comparable, just didn't want to deal with optionals since its not comparable.

Reply:
Yeah that makes sense

Reply:
Testing Maghrib after sunset is a safer test than after Asr, so I made the change. This is the details of the formula: http://praytimes.org/calculation